src/analysis/convex/segment.lean

```diff
@@ -306,6 +306,29 @@ begin
   rw [←sub_eq_neg_add, ←neg_sub, hxy, ←sub_eq_neg_add, hzx, smul_neg, smul_comm, neg_add_self]
 end
 
+open affine_map
+
+lemma open_segment_subset_union (c : 𝕜) (x y : E) :
+  open_segment 𝕜 x y ⊆ insert (line_map x y c) (open_segment 𝕜 x (line_map x y c) ∪
+    open_segment 𝕜 (line_map x y c) y) :=
```

Comment:
This statement could do with a docstring explanation.

EDIT: If you modify it according to my next comment, maybe not.